api/src/test/java/org/pmiops/workbench/actionaudit/auditors/SumoLogicAuditorTest.java

```diff
@@ -0,0 +1,216 @@
+        .containsExactly(WORKSPACE_ID);
+
+    // We should have distinct event rows with values from the egress event.
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventTargetProperty.EGRESS_MIB.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .collect(Collectors.toSet()))
+        .containsExactly("12.3");
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventTargetProperty.VM_NAME.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .collect(Collectors.toSet()))
+        .containsExactly(EGRESS_EVENT_VM_NAME);
+  }
+
+  @Test
+  public void testNoWorkspaceFound() {
+    exception.expect(BadRequestException.class);
+
+    // When the workspace lookup doesn't succeed, the event is filed w/ a system agent and an
+    // empty target ID.
+    when(mockWorkspaceService.getByNamespace(WORKSPACE_NAMESPACE)).thenReturn(null);
+    sumoLogicAuditor.fireEgressEvent(
+        new EgressEvent().projectName(EGRESS_EVENT_PROJECT_NAME).vmName(EGRESS_EVENT_VM_NAME));
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    // Some of the properties should be nulled out, since we can't identify the target workspace
+    // for the egress event.
+    assertThat(events.stream().map(event -> event.getAgentId()).collect(Collectors.toSet()))
+        .containsExactly(0);
+    assertThat(events.stream().map(event -> event.getAgentEmailMaybe()).collect(Collectors.toSet()))
+        .containsExactly((Object[]) null);
+    assertThat(events.stream().map(event -> event.getTargetIdMaybe()).collect(Collectors.toSet()))
+        .containsExactly((Object[]) null);
+
+    // We expect to see an audit event row with a comment describing the issue encountered when
+    // trying to handle the high-egress message.
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventCommentTargetProperty.COMMENT.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .findFirst()
+                .get())
+        .contains("Failed to find workspace");
+  }
+
+  @Test
+  public void testFailedParsing() {
+    // When the inbound request parsing fails, an event is logged at the system agent.
+    when(mockWorkspaceService.getByNamespace(WORKSPACE_NAMESPACE)).thenReturn(null);
+    sumoLogicAuditor.fireFailedToParseEgressEvent(new EgressEventRequest().eventsJsonArray("asdf"));
```

Comment:
nit: If the array isn't valid JSON, I would've expected a lower-level exception to be thrown first.

Reply:
🤷‍♂

Reply:
Except that your controller is passing it in internally, so this makes sense.